debezium-server/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java

```diff
@@ -0,0 +1,121 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpChangeConsumer.class);
+
+    private static final String PROP_PREFIX = "debezium.sink.http.";
+    private static final String PROP_WEBHOOK_URL = "url";
+    private static final String PROP_CLIENT_TIMEOUT = "timeout.ms";
+
+    private static final Long HTTP_TIMEOUT = Integer.toUnsignedLong(60000); // Default to 60s
+
+    private static String sinkUrl;
+    private static Duration timeoutDuration;
+    private static String contentType;
+
+    private HttpClient client;
+    private HttpRequest.Builder requestBuilder;
+
+    // If this is running as a Knative object, then expect the sink URL to be located in `K_SINK`
+    // as per https://knative.dev/development/eventing/custom-event-source/sinkbinding/
+    @PostConstruct
+    void connect() throws URISyntaxException {
+        client = HttpClient.newHttpClient();
+        final Config config = ConfigProvider.getConfig();
+        String sink = System.getenv("K_SINK");
+        timeoutDuration = Duration.ofMillis(HTTP_TIMEOUT);
+
+        if (sink != null) {
+            sinkUrl = sink;
+        }
+        else {
+            sinkUrl = config.getValue(PROP_PREFIX + PROP_WEBHOOK_URL, String.class);
+        }
+
+        config.getOptionalValue(PROP_PREFIX + PROP_CLIENT_TIMEOUT, String.class)
+                .ifPresent(t -> timeoutDuration = Duration.ofMillis(Long.parseLong(t)));
+
+        switch (config.getValue("debezium.format.value", String.class)) {
+            case "avro":
+                contentType = "avro/bytes";
+                break;
+            case "cloudevents":
+                contentType = "application/cloudevents+json";
+                break;
+            default:
+                // Note: will default to JSON if it cannot be determined, but should not reach this point
+                contentType = "application/json";
+        }
+
+        LOGGER.info("Using http content-type type {}", contentType);
+        LOGGER.info("Using sink URL: {}", sinkUrl);
+        requestBuilder = HttpRequest.newBuilder(new URI(sinkUrl)).timeout(timeoutDuration);
+        requestBuilder.setHeader("content-type", contentType);
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+            String value = "";
+            if (record.value() != null) {
+                value = (String) record.value();
+
+                HttpRequest request = requestBuilder.POST(HttpRequest.BodyPublishers.ofString(value)).build();
+                client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                        .thenAccept(r -> {
+                            if ((r.statusCode() == HTTP_OK) || (r.statusCode() == HTTP_NO_CONTENT)) {
```

Comment:
Knative uses [202](https://github.com/knative/specs/blob/main/specs/eventing/data-plane.md#event-acknowledgement-and-delivery-retry), can we add [`HTTP_ACCEPTED`](https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/net/HttpURLConnection.html#HTTP_ACCEPTED) too?

Reply:
I've seen Knative services use the above two, but how about if I were to verify it is inside the 2xx range instead?

Reply:
Sounds good!